Paybox/System/Base/Request.php

```diff
@@ -147,10 +148,23 @@ public function getForm($options = array())
         $options['csrf_protection'] = false;
 
         $parameters = $this->getParameters();
-        $builder = $this->factory->createNamedBuilder('', 'form', $parameters, $options);
-
-        foreach ($parameters as $key => $value) {
-            $builder->add($key, 'hidden');
+        // If symfony version is 3.* then we use the FQCN for form types
+        // Else we use the IDs.
+        if (substr(Kernel::VERSION, 0, 1) === "3") {
```

Comment:
Kernel version comparison is not good at all.

If you have a project not using SF full stack, it might be possible to have a kernel version different from Form component.

You should use code discovery instead (e.g. `method_exists`).